addons/auth/package.json

```diff
@@ -10,14 +10,14 @@
   "scripts": {
     "lint": "npm-run-all --aggregate-output --continue-on-error --parallel 'lint:!(fix)'",
     "lint:fix": "npm-run-all --aggregate-output --continue-on-error --parallel lint:*:fix",
-    "lint:hbs:fix": "ember-template-lint . --fix",
     "lint:js": "eslint .",
+    "lint:js:fix": "eslint . --fix",
     "format": "npm run format:js",
     "format:js": "prettier --write '{addon,app,config,stories,tests}/**/*.js' *.js",
-    "lint:js:fix": "eslint . --fix",
     "start": "ember serve",
     "test": "ember test",
     "test:ember-compatibility": "ember try:each",
+    "precommit": "yarn lint && yarn format && yarn test",
```

Comment:
How would this `precommit` work? 
I was under the impression our root `package.json` has a `precommit` script which is just an npm/yarn hook based solely on the name (side note, does anyone use yarn 2/will we need to migrate? It looks like yarn 2 [won't support these pre/post hooks](https://yarnpkg.com/getting-started/migration#explicitly-call-the-pre-and-post-scripts))

Another question is I notice `yarn format` would run `prettier --write` but would that stage the files even if this script ran? Have we considered using actual git hooks to run `eslint --fix` and `prettier --write` and staging them so we don't have to worry about fixing styles and failing a commit if it wasn't properly formatted?

Reply:
@ZedLi 

Excellent question!

- We use commitizen with default behaviour, this is what manages the `yarn commit`.
- The `precommit` script within root `package.json` it is the only one triggered by the hook (commitizen `git-cz` explained above).
- What `precommit` does you can follow the breadcrumbs but in a nutshell:
  - runs `compliance:license`.
  - runs yarn lint, which thanks to [npm-run-all](https://www.npmjs.com/package/npm-run-all) runs all the `lint:*` yarn scripts defined within the root `package.json`.
  - runs yarn test, as explained above, runs all the `test:` yarn scripts defined within the root `package.json`.

So the hook just exists within the root of the project. And the magic of it based on [cz-cli](https://github.com/commitizen/cz-cli.).

Any `precommit` script within the submodules will not be triggered by any hooks.

The goal of providing individual `precommit` scripts per project is to facilitate the developer work when just applying changes to a specific project. In example, you have been working within `addons/api` and before running the `yarn commit` from the root of the project , which takes a bit of time, you want to run such checks first just for `addons/api`, which will be way faster.

Reply:
@ZedLi 

As per your second question:
Feel free to bring this discussion to FE tech time. As per my understanding and usage of the `yarn format`. What it does is "fix" the format issues but does not stage files for you, just fix the issues. Is responsibility of the developer to stage and manage the changes with git.
If you run locally `yarn commit` or `yarn precommit` before pushing a branch the CI should run the exact same tools. But it is true, that if you commit with a classic `git commit` and does not run the `yarn commit` then no hooks will run at all.

Maybe the scripts and tools available for the dev should be better documented (working on it) but this is an interesting topic I would like to discuss over FE tech time

Reply:
It depends how we choose to setup the hooks I think, I know the `cloud-ui` repo is manual but I've setup husky in the past which has a `prepare` script (this might be different for yarn and wasn't even needed with earlier versions) which automatically setsup the hooks when you do an `npm install`. 

Generally that was "good enough" in that any new contributor would have the hooks setup when they install, but any current developer would have to still go through that install step either by manually setting them up or doing another `npm install`

Reply:
Ah that makes a lot of sense.  We could do a `postinstall` script!

Reply:
I think this is very interesting, despite this PR started this discussion, it goes a bit out of the scope.
For the sake of this PR and since we already start on FE tech time talks about it, I will resolve this conversation.
Feel free to reopen if you think so.